source/particle/generator/reference_cell.cc

```diff
@@ -0,0 +1,192 @@
+/*
+  Copyright (C) 2015 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#include <aspect/particle/generator/reference_cell.h>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Generator
+    {
+      template <int dim>
+      ReferenceCell<dim>::ReferenceCell()
+      {
+        starting_particle_index = 0;
+      }
+
+      template <int dim>
+      void
+      ReferenceCell<dim>::generate_particles(std::multimap<types::LevelInd, Particle<dim> > &particles)
+      {
+        const std::vector<Point<dim>> particles_in_unit_cell = generate_particle_positions_in_unit_cell();
```

Comment:
please put a space between the two closing `>` `>`, older compilers misinterpret this sometimes as a `>>` operator (we had this problem a while ago).